presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -273,7 +273,8 @@ private void processNewSources()
                 this.split.set(split);
             }
 
-            Supplier<Optional<UpdatablePageSource>> pageSource = sourceOperator.addSplit(split);
+            Supplier<Optional<UpdatablePageSource>> pageSource =
+                    sourceOperator instanceof WrapperSourceOperator ? ((WrapperSourceOperator) sourceOperator).addSplit(newSplit) : sourceOperator.addSplit(split);
```

Comment:
hmmm not necessary. Let's refactor the interface of `SourceOperator` to take `ScheduledSplit`